File: web/app/js/types/subscription.ts
import { Tag } from "./syndication";

export interface Subscription {
  id: string;
  url: URL;
  domain: URL | null;
  title: string;
  type: string;
  isSubscribed: boolean;
  frequency: string;
  createdAt: Date | null;
  updatedAt: Date | null;
}

export interface SubscriptionResults {
  limit: number;
  total: number;
  offset: number;
  results: Subscription[];
}

export interface SearchParams {
  terms: string[];
  tags: string[];
}

export interface TagResults {
  total: number;
  results: Tag[];
}

File: internal/domain/subscription/subscription.go
package subscription

import (
	"github.com/mickaelvieira/taipan/internal/domain/http"
	"github.com/mickaelvieira/taipan/internal/domain/syndication"
	"github.com/mickaelvieira/taipan/internal/domain/url"
	"time"
)

// Subscription represents a user'subscription to a syndication source
type Subscription struct {
	ID         string
	UserID     string
	URL        *url.URL
	Domain     *url.URL
	Type       syndication.Type
	Title      string
	Subscribed bool
	Frequency  http.Frequency
	CreatedAt  time.Time
	UpdatedAt  time.Time
}
